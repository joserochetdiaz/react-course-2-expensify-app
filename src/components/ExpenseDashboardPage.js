import React from 'React';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';


const ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)
export default ExpenseDashboardPage;