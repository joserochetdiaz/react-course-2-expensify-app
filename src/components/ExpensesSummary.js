import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => {
    const summary = selectExpensesTotal(props.expenses);
    const expenseWord = props.expenses.length === 1 ? 'expense' : 'expenses';
    return (
        <div>
            
            {props.expenses.length > 0 && <h3>You are now viewing {props.expenses.length} {expenseWord} totalling {numeral(summary).format('$0,0.00')}</h3>}
        </div>
      
    );
};
const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filters)
});
export default connect(mapStateToProps)(ExpensesSummary);