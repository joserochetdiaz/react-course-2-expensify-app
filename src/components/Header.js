import React from 'react';
import {NavLink} from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogOut } from '../actions/auth';

export const Header = ({ startLogOut }) => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="isActive" exact={true}>Home</NavLink>
        <NavLink to="/create" activeClassName="isActive">Create expense</NavLink>
        <NavLink to="/help">Help</NavLink>
        <button onClick={startLogOut}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogOut: () => dispatch(startLogOut())
});

export default connect(undefined, mapDispatchToProps)(Header);