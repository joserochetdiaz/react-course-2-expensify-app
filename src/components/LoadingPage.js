import React from 'react';

const LoadingPage = () => (
    <div>
        <img src="../public/images/loader.gif"/>
    </div>
);

export default LoadingPage;