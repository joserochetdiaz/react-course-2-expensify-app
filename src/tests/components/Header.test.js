import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import {Header} from '../../components/Header';

test('Should render Header correctly', () => {
    const wrapper = shallow(<Header startLogOut={() => {}}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
});

test('should call logout function on button click', () => {
    const startLogOut = jest.fn();
    const wrapper = shallow(<Header startLogOut = {startLogOut}/>);
    wrapper.find('button').simulate('click');
    expect(startLogOut).toHaveBeenCalled();
});