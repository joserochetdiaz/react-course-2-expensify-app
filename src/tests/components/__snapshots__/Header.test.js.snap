// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="isActive"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="isActive"
    to="/create"
  >
    Create expense
  </NavLink>
  <NavLink
    to="/help"
  >
    Help
  </NavLink>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
